pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -121,27 +122,38 @@ func (c *Cluster) configureDashboardModule(m *mgrConfig) error {
 		return err
 	}
 
-	// server port
-	port := strconv.Itoa(m.DashboardPort)
-	changed, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
-	if err != nil {
-		return err
-	}
-	hasChanged = hasChanged || changed
-
 	// ssl support
 	var ssl string
 	if c.dashboard.SSL == nil {
 		ssl = ""
 	} else {
 		ssl = strconv.FormatBool(*c.dashboard.SSL)
 	}
-	changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
+	changed, err := client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/ssl", ssl, false)
 	if err != nil {
 		return err
 	}
 	hasChanged = hasChanged || changed
 
+	// server port
+	port := strconv.Itoa(m.DashboardPort)
+	changed, err = client.MgrSetConfig(c.context, c.Namespace, m.DaemonID, c.clusterInfo.CephVersion, "mgr/dashboard/server_port", port, false)
+	if err != nil {
+		return err
+	}
+	hasChanged = hasChanged || changed
+
+	// SSL enabled. Needed to set specifically the ssl port setting starting with Nautilus(14.2.1)
+	if ssl == "true" {
```

Comment:
The string variable for `ssl` was only needed for setting the config option above. Since this is a bool you could check `if c.dashboard.SSL` here.

Reply:
Modified following your suggestion.

Using <c.dashboard.SSL> in the conditional was my first try. But looking at the conversion to string of this variable at the beginning i had doubts. (in what cases c.dashboard.SSL can be nil?)

I have simplified the <ssl> assignation taking into account that if there is no initialization of <c.dashboard.SSL> golang will assign the default value (False) to it.

I can't see when this attribute can be set to a nil value but my knowledge of the language and the project is still very light. 
Is right my assumption about that no initialization of <c.dashboard.SSL> will result in assign false by default to the attribute?

Reply:
@travisn , @leseb :
Trying to find why is needed to check if <c.dashboard.SSL> is nil , i have found that it is declared in the <DashboardSpec> as *bool. 
Now i do not understand why this attribute is declared as a pointer instead of being declared directly as bool ....  8-|
Any help to understand this will be welcome, maybe the declaration is wrong and we should change to bool... seems very strange

Reply:
I have no idea why we had a pointer here, but your current change looks good to me.

Reply:
I think it was a pointer so that the default value would be `nil`, and `nil` was interpreted as enabled. With the value just as `bool` the default becomes `false` which means disabled, so this changes the default.

Reply:
Make senses, we should probably uncomment this https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/cluster.yaml#L50 then.

Reply:
created #3862